src/components/POS/Classic/SelectedItemRow.vue

```diff
@@ -181,7 +182,7 @@
         :show-label="true"
         :border="true"
         :value="row.itemDiscountAmount"
-        :read-only="row.itemDiscountPercent as number > 0 || isReadOnly"
+        :read-only="row.itemDiscountPercent as number > 0 && isRateReadOnly()"
```

Comment:
No need to use isRateReadOnly() on the discount field

```suggestion
        :read-only="isDiscountsReadOnly(row.itemDiscountPercent as number < 0)"
```